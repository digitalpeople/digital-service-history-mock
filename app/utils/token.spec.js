import jwt from 'jsonwebtoken';
import createToken from './token';

jest.mock('jsonwebtoken', () => ({
  sign: jest.fn(),
}));

describe('createToken', () => {
  it('should call jwt.sign', () => {
    expect.assertions(1);

    const content = {
      username: 'mock-username',
    };

    createToken(content);

    expect(jwt.sign.mock.calls[0]).toMatchSnapshot();
  });
});
