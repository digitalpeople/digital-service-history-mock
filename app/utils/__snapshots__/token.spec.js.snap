// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createToken should call jwt.sign 1`] = `
Array [
  Object {
    "username": "mock-username",
  },
  "mock jwt key",
  Object {
    "expiresIn": "1h",
  },
]
`;
